### PR TITLE
enhancement(cocoapods): Remove version pin

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Enhancement**

### Description

Also came out of the RN upgrade attempt.

Cocoapods stays stable mostly, and every now and then we need to update, as with fastlane. Both upgrade more easily when there is no pin, with just `bundle update cocoapods`, whenever there is the need for it. For RN 0.64 there was.

Pinning doesn't really help us for fastlane and cocoapods, where we just want the latest version that works. If we ever get to the point where we don't want some specific new version, then we can pin again.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434

### CODE REVIEW

File: Artsy.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3311,6 +3311,8 @@
 				"${PODS_ROOT}/Artsy+UIFonts/Pod/Assets/Unica77LL-Regular.otf",
 				"${PODS_ROOT}/Artsy+UILabels/Pod/Assets/Chevron_Black.png",
 				"${PODS_ROOT}/Artsy+UILabels/Pod/Assets/Chevron_Black@2x.png",
+				"${PODS_ROOT}/../emission/Pod/Assets/Emission.js",
+				"${PODS_ROOT}/../emission/Pod/Assets/assets",
```
Comment (ds300) [inline Artsy.xcodeproj/project.pbxproj:3315]: I think the changes in this file shouldn't be staged because it makes building the app fail for people who haven't run `react-native bundle` locally. I could be wrong 🤔

Are these changes intentional? If so can you try checking out this branch, removing `emission/Pod/Assets`, and building the app for dev mode?

Comment (pvinis) [reply Artsy.xcodeproj/project.pbxproj:3315]: It appeared after running `bundle e pod install`, so I guessed it was needed. I can remove them and if they appear for others too, we commit them then? 🤔

Comment (pvinis) [reply Artsy.xcodeproj/project.pbxproj:3315]: I'll try what you said now

Comment (pvinis) [reply Artsy.xcodeproj/project.pbxproj:3315]: oh yea. interesting. ok removing this file change then. can't wait till we fix that thing with emission.
